view/frontend/web/js/model/spi.js

```diff
@@ -71,6 +85,10 @@ define([
                 ],
                 'callbacks': {
                     'onCreatePaymentOrder': async (paymentType, paymentPayload) => {
+                        if (!validateAgreements()) {
+                            throw new Error('Agreements not accepted');
+                        }
+
```

Comment:
This change should also be added to the `onUpdatePaymentOrder` callback as well (if applicable), since Apple Pay and Google call that one _before_ `onCreatePaymentOrder`.